tracer/src/Datadog.Trace/Debugger/Symbols/SymbolBatchUploadApi.cs

```diff
@@ -0,0 +1,121 @@
+// <copyright file="SymbolBatchUploadApi.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.DiscoveryService;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Debugger.Sink;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Symbols
+{
+    internal class SymbolBatchUploadApi : IBatchUploadApi
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<SymbolBatchUploadApi>();
+        private readonly IApiRequestFactory _apiRequestFactory;
+        private readonly ArraySegment<byte> _eventMetadata;
+        private string? _endpoint;
+        private IDiscoveryService? _discoveryService;
+
+        private SymbolBatchUploadApi(IApiRequestFactory apiRequestFactory, IDiscoveryService discoveryService, ArraySegment<byte> eventMetadataMetadata)
+        {
+            _apiRequestFactory = apiRequestFactory;
+            _eventMetadata = eventMetadataMetadata;
+            _discoveryService = discoveryService;
+            _discoveryService.SubscribeToChanges(ConfigurationChanged);
+        }
+
+        private static ArraySegment<byte> GetEventAsArraySegment(string? serviceName)
```

Comment:
`GetEventMetadataAsArraySegment`

Reply:
Fixed in db6086b623f7ad8ae43181865be84140bce1ef3c